forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt

```diff
@@ -33,22 +45,80 @@ data class FROptions(val server: Server,
                     && old?.urlPath == new?.urlPath
                     && old?.logger == new?.logger
         }
+
+
     }
+
     @Throws(IllegalArgumentException::class)
     @JvmName("validateConfig")
     internal fun validateConfig() {
         require(server.url.isNotBlank()) { "AM URL cannot be blank" }
         require(server.realm.isNotBlank()) { "Realm cannot be blank" }
         require(server.cookieName.isNotBlank()) { "cookieName cannot be blank" }
     }
+
+    private fun getOkHttpClient(url: URL): OkHttpClient {
+        val networkConfig = NetworkConfig.networkBuilder()
+            .timeUnit(TimeUnit.SECONDS)
+            .host(url.authority)
+            .interceptorSupplier {
+                listOf<Interceptor>(
+                    OkHttpRequestInterceptor()
+                )
+            }
+            .build()
+
+        // Obtain instance of OkHttp client
+        val httpClient = getInstance().lookup(networkConfig)
+        return httpClient
+    }
+
+    suspend fun discover(discoverUrl: String): FROptions {
+        return withContext(Dispatchers.IO) {
+            suspendCancellableCoroutine { continuation ->
```

Comment:
I think we don't need to use suspendCancellableCoroutine, may just invoke httpClient.newCall().execute?